### ISSUE TITLE
docstring: Regarding Korean line spacing problem

### ISSUE THREAD

**Issue by swdev-j** [microsoft/pylance-release#4615]
## Environment data

-   Language Server version: v2023.7.21
-   OS and version: windows 11
-   Python version (& distribution if applicable, e.g. Anaconda): 3.9.12

## Code Snippet

```python
# -*- coding: utf-8 -*-

def fun(a:int, b:int, c:int):
    """_summary_
    test
    
    Args:
        a (int): _description_
        b (int): 한국어
        c (int): _description_
    """

print(fun.__doc__)
```

# Repro Steps

1. write and standard docstring that contains parenthesis and then hover over the function to observe the formatting of the docstring.
2. compare this to the expected formatting given by print(my_function.__doc__).

## Expected behavior

I believe this will output something like below.

![image](https://github.com/microsoft/pylance-release/assets/82888550/848845ad-30a2-4f6b-9dea-fb9ec8753868)

## Actual behavior

But, I hover over the function, it does not same output.
It should show the same result as print.

![image](https://github.com/microsoft/pylance-release/assets/82888550/0e9ffc38-2732-401d-a0b7-28beef5f9dd6)

**Comment by heejaechang** [comment]
looks like our regex for doc string doesn't work properly with Korean. (or any other language except English)

**Comment by heejaechang** [comment]
tagging @bill ?

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2024.6.100, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20246100-13-June-2024-PreRelease)

**Comment by rchiodo** [comment]
The fix for this issue is behind an experimental feature flag:

```json
"python.analysis.supportRestructuredText": true
```

If you want to try out our new restructuredText support, enable this flag. It's behind a flag at the moment until we can make sure it handles all the possible Sphinx/GoogleDoc/Epytext scenarios that customers need. Please log additional issues if this setting isn't working out for you.

Thanks :)
